packages/@glimmer/runtime/lib/managers/index.ts

```diff
@@ -0,0 +1,236 @@
+
+///////////
+
+const getPrototypeOf = Object.getPrototypeOf;
+
+function setManager<O extends Owner, Def extends object>(
+  map: WeakMap<Def, ManagerFactory<O>>,
+  factory: ManagerFactory<O> | ManagerFactory<O | undefined>,
+  obj: Def
+): Def {
+  if (DEBUG && (typeof obj !== 'object' || obj === null) && typeof obj !== 'function') {
+    throw new Error(
+      `Attempted to set a manager on a non-object value. Managers can only be associated with objects or functions. Value was ${debugToString!(
+        obj
+      )}`
+    );
+  }
+
+  if (DEBUG && map.has(obj)) {
+    throw new Error(
+      `Attempted to set the same type of manager multiple times on a value. You can only associate one manager of each type with a given value. Value was ${debugToString!(
+        obj
+      )}`
+    );
+  }
+
+  map.set(obj, factory);
+  return obj;
+}
+
+function getManager<O, D extends ManagerDelegate>(
+  map: WeakMap<object, ManagerFactory<O, D>>,
+  obj: object
+): ManagerFactory<O, D> | undefined {
+  let pointer = obj;
+  while (pointer !== undefined && pointer !== null) {
+    const manager = map.get(pointer);
+
+    if (manager !== undefined) {
+      return manager;
+    }
+
+    pointer = getPrototypeOf(pointer);
+  }
+
+  return undefined;
+}
+
+function getManagerInstanceForOwner<D extends ManagerDelegate>(
+  owner: Owner | undefined,
+  factory: ManagerFactory<Owner, D>
+): D {
+  let managers;
+
+  if (owner === undefined) {
+    managers = UNDEFINED_MANAGER_INSTANCES;
+  } else {
+    managers = OWNER_MANAGER_INSTANCES.get(owner);
+
+    if (managers === undefined) {
+      managers = new WeakMap();
+      OWNER_MANAGER_INSTANCES.set(owner, managers);
+    }
+  }
+
+  let instance = managers.get(factory);
+
+  if (instance === undefined) {
+    instance = factory(owner!);
+    managers.set(factory, instance!);
+  }
+
+  // We know for sure that it's the correct type at this point, but TS can't know
+  return instance as D;
+}
+
+///////////
+
+export function setModifierManager<O extends Owner>(
+  factory: ManagerFactory<O, ModifierManager<unknown> | InternalModifierManager>,
+  definition: object
+) {
+  return setManager(MODIFIER_MANAGERS, factory, definition);
+}
+
+export function getModifierManager(
+  owner: Owner | undefined,
+  definition: object
+): ModifierManager<unknown> | InternalModifierManager | undefined {
+  const factory = getManager(MODIFIER_MANAGERS, definition);
+
+  if (factory !== undefined) {
+    let manager = getManagerInstanceForOwner(owner, factory);
+
+    if (
+      DEBUG &&
+      !isInternalModifierManager(manager) &&
+      !FROM_CAPABILITIES!.has(manager.capabilities)
+    ) {
+      throw new Error(
+        `Custom modifier managers must have a \`capabilities\` property that is the result of calling the \`capabilities('3.13' | '3.22')\` (imported via \`import { capabilities } from '@ember/modifier';\`). Received: \`${JSON.stringify(
```

Comment:
Assuming that Glimmer.js will also leverage this infrastructure (I assume it has to), we need to allow this suggested string to be "host dependent". Can we have this configured by the host instead?

Obviously, we should also work to make the value the same (so that both Ember and Glimmer.js apps import from the same place), but we'll need an RFC for that...

Reply:
IMHO, making the error message _less_ useful is not a valid choice here, I think if we have to land this as `@ember/modifier` in the string for now that is "fine" (by some definition 😝 ) but we should make an issue (and an inline comment pointing to that issue) to come back and clean this up (either by allowing the host to customize the value, or by doing the RFC in Ember land to allow the import to be from an `@glimmer` namespace).